lib/rogue.js

```diff
@@ -1,136 +1,54 @@
-}
-/**
- * executePost
- *
- * @param {string} endpoint
- * @param {Object} data - payload to be sent in the POST request to Rogue
- * @param {string} version = 'v2' - Rogue API version
- * @param {string} strategyName = 'apiKeyCredentials'
- *                                Auth strategy used to send requests to Rogue's API
- */
-function executePost(endpoint, data, version = 'v2', strategyName = 'apiKeyCredentials') {
-  return rogueClient
-    .request(strategyName)
-    .post(getRogueUrl(endpoint, version))
-    .send(data)
-    .then(res => res.body);
-}
-/**
- * executeV3PostsPost
- *
- * @param {string} endpoint
- * @param {Object} data - payload to be sent in the POST request to Rogue
- */
-function executeV3PostsPost(endpoint, data) {
-  const fileProperty = rogueClientConfig.photoPostCreation.fileProperty;
-  const client = rogueClient.request('clientCredentials').post(getRogueUrl(endpoint, 'v3'));
-  /**
-   * We need to setup the superagent request differently for V3 POST /posts requests.
-   * The request has to be of Content-Type: multipart/form-data, but it must not be set manually.
-   * It detects it by using the .field and .attach API methods.
-   */
-  Object.keys(data).forEach((key) => {
-    if (key === fileProperty) {
-      client.attach(fileProperty, data[key], 'post-photo');
-    } else {
-      client.field(key, data[key]);
-    }
-  });
-  /**
-   * In the future, Rogue is implementing the concept of actions, for now this
-   * is the expected action
-   */
-  client.field('action', 'default');
-  return client.then(res => res.body);
+let rogueClient;
+
+function getClient() {
+  if (!rogueClient) {
+    rogueClient = RogueClient.getNewInstance();
+  }
+  return rogueClient;
 }
 /**
  * createPost
  *
  * @param {object} data
  * @return {Promise}
  */
-module.exports.createPost = function (data) {
-  // TODO: remove conditional when we fully switch to Rogue V3
-  if (rogueClientConfig.useV3()) {
-    return executeV3PostsPost('posts', data, 'v3', 'clientCredentials');
-  }
-  // TODO: Remove when we fully switch to Rogue V3
-  return executePost('posts', data);
-};
+function createPost(data) {
+  return getClient().Posts.create(data);
+}
 /**
  * createSignup
  *
  * @param {object} data
  * @return {Promise}
  */
-module.exports.createSignup = function (data) {
-  // TODO: remove conditional when we fully switch to Rogue V3
-  if (rogueClientConfig.useV3()) {
-    return executePost('signups', data, 'v3', 'clientCredentials');
-  }
-  // TODO: Remove when we fully switch to Rogue V3
-  return executePost('signups', data);
-};
+function createSignup(data) {
+  return getClient().Signups.create(data);
+}
 /**
  * fetchActivityForUserIdAndCampaignRunId
  *
  * @param {string} userId
  * @param {number} campaignRunId
  * @return {Promise}
  */
-module.exports.fetchActivityForUserIdAndCampaignRunId = function (userId, campaignRunId) {
+function fetchActivityForUserIdAndCampaignRunId(userId, campaignRunId) {
```

Comment:
Minor TODO for cleanup later here I think would be to rename to `getByUserIdAndCampaignRunId`